presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -605,7 +606,12 @@ public void run()
                         }
                     }
                     catch (Throwable t) {
-                        log.error(t, "Error processing %s", split.getInfo());
+                        if (t instanceof PrestoException) {
+                            log.error("Error processing %s: %s", split.getInfo(), t);
```

Comment:
You might want to call `t.getMessage()` instead of letting the logger call `t.toString()`. The `toString` of an exception contains the exception class name, which in this case is a bit redundant.